common/src/main/java/org/jboss/hal/testsuite/creaper/command/AddSocketBinding.java

```diff
@@ -0,0 +1,156 @@
+package org.jboss.hal.testsuite.creaper.command;
+
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author Jan Kasik <jkasik@redhat.com>
+ *         Created on 19.11.15.
+ */
+public class AddSocketBinding implements OnlineCommand {
+
+    private final String socketBindingName;
+
+    private final List<String> clientMappings;
+    private final Boolean fixedPort;
+    private final String interfaceName;
+    private final String multicastAddress;
+    private final Integer multicastPort;
+    private final Integer port;
+    private final boolean replaceExisting;
+
+    private AddSocketBinding(Builder builder) {
+        this.socketBindingName = builder.socketBindingName;
+        this.clientMappings = builder.clientMappings;
+        this.fixedPort = builder.fixedPort;
+        this.interfaceName = builder.interfaceName;
+        this.multicastAddress = builder.multicastAddress;
+        this.multicastPort = builder.multicastPort;
+        this.port = builder.port;
+        this.replaceExisting = builder.replaceExisting;
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws IOException {
+        Operations ops = new Operations(ctx.client);
+
+        String socketBindingGroup = ctx.client.options().isDomain ? "full-sockets" : "standard-sockets";
+        Address socketBindingAddress = Address.of("socket-binding-group", socketBindingGroup)
+                .and("socket-binding", socketBindingName);
+
+        if (replaceExisting) {
+            try {
+                ops.removeIfExists(socketBindingAddress);
+            } catch (OperationException e) {
```

Comment:
Thanks, I wrapped it as CommandFailedException instead, i think I chose IOException because it was already present because of `ops.add...` but this gives more sense...